gateway/middleware.go

```diff
@@ -400,7 +400,7 @@ func (t BaseMiddleware) ApplyPolicies(session *user.SessionState) error {
 		policyIDs = session.PolicyIDs()
 		t.Gw.policiesMu.RLock()
 		lookupMap = t.Gw.policiesByID
-		defer t.Gw.policiesMu.RUnlock()
+		t.Gw.policiesMu.RUnlock()
```

Comment:
lookupMap is being accessed for the rest of the function, defer protects reading from the map